src/evalfunc.c

```diff
@@ -3776,6 +3778,7 @@ f_confirm(typval_T *argvars UNUSED, typval_T *rettv UNUSED)
 	return;
 
     message = tv_get_string_chk(&argvars[0]);
+    set_prompt(message);
```

Comment:
This should come after the NULL check, otherwise it will crash on type error.

Reply:
Fixed.